### ISSUE TITLE
Bug in Bulk Create

### ISSUE THREAD

**Issue by jpmateo022** [MasoniteFramework/orm#337]
I tried passing a list with a dictionary in creating a method to test the bulk create but its throwing this error.

```
dictionary update sequence element #0 has length 13; 2 is required
```

Example Code
```
            Devices.create([
                {
                    "device_id": "device-c5fa4a62f740129966a9434556e16c0d", 
                    "hostname": "localhost1", 
                    "ip_address": "127.0.0.1", 
                    "os": "iosxr", 
                    "port": 90, 
                    "protocol": "ssh", 
                    "username": "asdasd", 
                    "password": "adasdasdasdasdasdasd", 
                    "queue_channel": "channel_3",
                    "proxy": "localhost", 
                    "is_proxy": 0, 
                    "type": "router", 
                    "created_by": "JP Mateo"
                }
            ])
```

**Comment by girardinsamuel** [comment]
Could you give us those details ?
- your devices create table migration
- your Device model file (at least the attributes you defined on the model e.g. `__fillable__`)

Thanks 👍

**Comment by jpmateo022** [comment]
I think it only accepts two keys per dictionary. I also tried the fillable variable and still failed.

Scenario 1: Two keys per dictionary in a list
Success:
```
            builder.create([{
                    'device_id': "asdasdasdasd",
                    'hostname':"Asdasdasdasd"
                },{
                    'device_id': "asdasdasdasd222",
                    'hostname':"Asdasdasdasd"
                }
            ])
```
![image](https://user-images.githubusercontent.com/26633918/105828904-a638bc00-5ffe-11eb-99fa-314ad350aed3.png)

Scenario 2: More than two  keys per dictionary in a list
Failed
```

            builder.create([{
                    'device_id': "asdasdasdasd",
                    'hostname':"Asdasdasdasd",
                    "ip_address":"127.0.0.1"
                },{
                    'device_id': "asdasdasdasd222",
                    'hostname':"Asdasdasdasd",
                    "ip_address":"127.0.0.2"
                }
            ])
```
![image](https://user-images.githubusercontent.com/26633918/105828982-c23c5d80-5ffe-11eb-9c01-dc0e4097e228.png)

**Comment by girardinsamuel** [comment]
Yeah that's the issue, that is why I was wondering what did you put in `__fillable__` parameters of your model ? You should put all fields that you want to use when calling create, it's mandatory.

**Comment by jpmateo022** [comment]
Yep I put all the fields in the __fillable__ then made the code Device.create()
```
__fillable__ = ["device_id","hostname","ip_address","os","port","protocol","username","password","queue_channel","proxy","is_proxy","type","created_by"]
```

Then the error message changed
```
message: "list indices must be integers or slices, not str"
```

**Comment by jpmateo022** [comment]
I tried inserting the list items individually and it work.

**Comment by girardinsamuel** [comment]
Sorry I did not catch that before but : 
- on an ORM model you should use: `Devices.bulk_create([])`
- on the query builder you should use: `builder.create([])`

Could you try 😉  ?

**Comment by jpmateo022** [comment]
Its kinda weird that its throws different errors.

for ORM
```
            Devices.bulk_create([{
                    'device_id': "asdasdasdasd",
                    'hostname':"Asdasdasdasd",
                    "ip_address":"127.0.0.1"
                },{
                    'device_id': "asdasdasdasd222",
                    'hostname':"Asdasdasdasd",
                    "ip_address":"127.0.0.2"
                }
            ])
```
message: "not all arguments converted during string formatting"

for Query Builder
```
            builder = QueryBuilder(table=Devices.__table__)
            builder.create([{
                    'device_id': "asdasdasdasd",
                    'hostname':"Asdasdasdasd",
                    "ip_address":"127.0.0.1"
                },{
                    'device_id': "asdasdasdasd222",
                    'hostname':"Asdasdasdasd",
                    "ip_address":"127.0.0.2"
                }
            ])
```
message: "dictionary update sequence element #0 has length 13; 2 is required"

**Comment by girardinsamuel** [comment]
Yeah that's weird, for the second one did you let 
```
__fillable__ = ["device_id","hostname","ip_address","os","port","protocol","username","password","queue_channel","proxy","is_proxy","type","created_by"]
```
on your model ? because it looks that the creation is authorized only with two fields.

**Comment by girardinsamuel** [comment]
@josephmancuso could you confirm this is the correct usage ? 
> for ORM
> 
> ```
>             Devices.bulk_create([{
>                     'device_id': "asdasdasdasd",
>                     'hostname':"Asdasdasdasd",
>                     "ip_address":"127.0.0.1"
>                 },{
>                     'device_id': "asdasdasdasd222",
>                     'hostname':"Asdasdasdasd",
>                     "ip_address":"127.0.0.2"
>                 }
>             ])
> ```
> 
> message: "not all arguments converted during string formatting"
> 
> for Query Builder
> 
> ```
>             builder = QueryBuilder(table=Devices.__table__)
>             builder.create([{
>                     'device_id': "asdasdasdasd",
>                     'hostname':"Asdasdasdasd",
>                     "ip_address":"127.0.0.1"
>                 },{
>                     'device_id': "asdasdasdasd222",
>                     'hostname':"Asdasdasdasd",
>                     "ip_address":"127.0.0.2"
>                 }
>             ])
> ```
> 
> message: "dictionary update sequence element #0 has length 13; 2 is required"

**Comment by jpmateo022** [comment]
For bulk_create at QueryBuilder.py:

It seems the bug is happening in this block of code.

```
        if not self.dry:
            connection = self.new_connection()
            query_result = connection.query(self.to_qmark(), self._bindings, results=1)

            processed_results = query_result or self._creates
        else:
            processed_results = self._creates
```

So my work around for now is this:

```
            sql = builder.bulk_create(data,query=True).to_sql()
            builder.statement(sql)
```

**Comment by josephmancuso** [comment]
I figured out the issue. I was testing on sqlite and this looks like an issue with MySQL (and probably postgres)

This is fixed in version 1.0.11.

Also there was an issue with the documentation as well.

```python
.create([{
                    'device_id': "asdasdasdasd",
                    'hostname':"Asdasdasdasd",
                    "ip_address":"127.0.0.1"
                },{
                    'device_id': "asdasdasdasd222",
                    'hostname':"Asdasdasdasd",
                    "ip_address":"127.0.0.2"
                }
            ])
```

is NOT the proper usage of bulk creating. It was documented wrong. It should be `bulk_create`

```python
.bulk_create([{
                    'device_id': "asdasdasdasd",
                    'hostname':"Asdasdasdasd",
                    "ip_address":"127.0.0.1"
                },{
                    'device_id': "asdasdasdasd222",
                    'hostname':"Asdasdasdasd",
                    "ip_address":"127.0.0.2"
                }
            ])
```

This is fixed in v1.0.11. This is what you get for coding and documenting at midnight 😅
